llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -1736,23 +1739,64 @@ static Value *generateNewInstTree(ArrayRef<InstLane> Item, FixedVectorType *Ty,
     Ops[Idx] = generateNewInstTree(generateInstLaneVectorFromOperand(Item, Idx),
                                    Ty, IdentityLeafs, SplatLeafs, Builder);
   }
+
+  FastMathFlags FMF;
+  FMF.setFast();
+  for_each(Item, [&FMF, Lane = FrontLane](const InstLane &E) {
+    if (E.second != Lane)
+      return;
+    auto *I = cast<Instruction>(E.first);
+    if (isa<FPMathOperator>(I))
```

Comment:
`if (auto *I = dyn_cast_or_null<FPMathOperator>(E.first))`, and I think it can drop the `E.second != Lane`, which I would guess is guarding against undef lanes where the value will be nullptr?

Reply:
I added `E.second != Lane` as `Item` contains the instructions for all lanes, so we can skip the ones that we have already seen before. Removing it would have the same effect, since we are still taking the logical and of all flags.